src/windows/main_window.py

```diff
@@ -1360,10 +1360,12 @@ def getTimelineObjectPositions(obj):
 
             fps = get_app().project.get("fps")
             fps_float = float(fps["num"]) / float(fps["den"])
+            frame_duration = float(fps["den"]) / float(fps["num"])
 
             clip_start_time = obj.data["position"]
             clip_orig_time = clip_start_time - obj.data["start"]
-            clip_stop_time = clip_orig_time + obj.data["end"]
+            # Last frame on clip is -1 frame's duration
+            clip_stop_time = clip_orig_time + obj.data["end"] - frame_duration
```

Comment:
Looks good! Walked through the logic with backwards and slowed down clips, and it makes sense.